### PR TITLE
Allow using callbacks as a data value.

### PR DESCRIPTION
**Pull request by mgralikowski**
Allow using callbacks as a data value.

### CODE REVIEW

**Comment by mgralikowski** [comment]
Sorry for the short description, but I was sure that I make full described PR with a use case and it disappeared or simply I forgot to click "Save"..

File: src/Html/Builder.php [hunk 1]
```diff
@@ -260,7 +260,8 @@ protected function makeDataScript(array $data)
     {
         $script = '';
         foreach ($data as $key => $value) {
-            $script .= PHP_EOL . "data.{$key} = '{$value}';";
+            $dataValue = $this->isCallbackFunction($value, $key) ? $value : "'{$value}'";
```
Comment (yajra) [inline src/Html/Builder.php:263]: Maybe the semi-colon should be placed here? `"'{$value}';"`

File: src/Html/Builder.php [hunk 2]
```diff
@@ -260,7 +260,8 @@ protected function makeDataScript(array $data)
...
+            $script .= PHP_EOL . "data.{$key} = {$dataValue};";
         }
```
Comment (yajra) [inline src/Html/Builder.php:265]: And remove the semi-colon here. `"data.{$key} = {$dataValue}";`

**Comment by mgralikowski** [comment]
Generally this code is tested, with callback and string as well. We need
semicolon to separate (js allow to ommit but I don't like do it) next
variable. I would to improve this code with handling types - now, int is
converting to string.

śr., 20 maj 2020, 04:04 użytkownik Arjay Angeles <notifications@github.com>
napisał:

**Review by yajra — APPROVED** [review]

**Comment by yajra** [comment]
Sorry for overlooking this PR. Released on v4.30.2. Thanks!

**Pull request merged**